### PR TITLE
feat(core): normalized system messages using elgg_view_message

### LINKED ISSUE #9880: Formalize using status messages in content
There are cases where I want (colored) status messages in the page content, and which do not disappear or have a pointer cursor. I guess I want to be able to put `<p class="elgg-message elgg-state-notice">` into content without fear of behavior/positioning being broken by a plugin.

I guess I'm asking to move `cursor:pointer` out of `.elgg-message { }` and add some comments to discourage plugin devs from applying positioning/behavior based on this class.

### PR DESCRIPTION
**Pull request by jdalsem**
fixes #9880 

meant to introduce a simple way to use messages in content (and anywhere else on the page)

- [x] upgrade notice
- [x] document function in docs
- [x] add icon support in title

image

### CODE REVIEW

File: views/default/admin/configure_utilities/robots.php [hunk 1]
```diff
@@ -3,15 +3,13 @@
  * Set robots.txt
  */
 
-if ('/' !== parse_url(elgg_get_site_url(), PHP_URL_PATH)) {
-	$warning = elgg_echo('admin:robots.txt:subdir');
-	echo "<div class=\"elgg-admin-notices\"><p>$warning</p></div>";
-}
+// if ('/' !== parse_url(elgg_get_site_url(), PHP_URL_PATH)) {
```
Comment (jdalsem) [inline views/default/admin/configure_utilities/robots.php]: don't forget to revert

**Comment by hypeJunction** [comment]
Dismiss option would be nice

File: engine/lib/views.php [hunk 2]
```diff
@@ -1210,6 +1210,26 @@ function elgg_view_module($type, $title, $body, array $vars = []) {
...
 }
 
+/**
+ * Wrapper function for the message display pattern.
+ *
+ * Box with header, body
+ *
+ * This is a shortcut for {@elgg_view page/components/message}.
+ *
+ * @param string $type  The type of message (main, info, popup, aside, etc.)
```
Comment (jeabakker) [inline engine/lib/views.php]: too much copy&paste from elgg_view_module `(main, info, popup, aside, etc.)` whould be `(success, warning, error, etc.)`

File: views/default/core/js/ui.js [hunk 3]
```diff
@@ -20,8 +20,8 @@ elgg.ui.init = function () {
 		}
 	});
 
-	$('.elgg-system-messages li').animate({opacity: 0.9}, 6000);
-	$('.elgg-system-messages li.elgg-state-success').fadeOut('slow');
+	$('.elgg-page-default .elgg-system-messages .elgg-message').parent().animate({opacity: 0.9}, 6000);
```
Comment (iionly) [inline views/default/core/js/ui.js]: Any chance to make the duration time configurable? If I'm not mistaken there was even an issue about that once (not sure if it's still open).

File: views/default/page/components/message.php [hunk 4]
```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * Elgg message element
+ *
+ * @uses $vars['type']      The type of message (error, success, warning, help, notice)
+ * @uses $vars['title']     Optional title text, will default to the type
+ * @uses $vars['icon_name'] Optional iconname to override default icon
```
Comment (hypeJunction) [inline views/default/page/components/message.php]: `icon` would be consistent with other views

File: views/default/page/elements/messages.php [hunk 5]
```diff
@@ -9,16 +9,14 @@
...
-if (isset($vars['object']) && is_array($vars['object']) && sizeof($vars['object']) > 0) {
-	foreach ($vars['object'] as $type => $list) {
-		foreach ($list as $message) {
-			$list_items .= elgg_format_element('li', [
-				'class' => "elgg-message elgg-state-$type",
-			], elgg_autop($message));
-		}
+foreach ($messages as $type => $list) {
+	foreach ($list as $message) {
+		$list_items .= elgg_format_element('li', [], elgg_view_message($type, $message, ['title' => false]));
```
Comment (mrclay) [inline views/default/page/elements/messages.php]: The single array syntax would be clearer.

File: views/default/core/js/ui.js (hunk 3 above)
Comment (jdalsem) [reply views/default/core/js/ui.js]: could be... but that should be discussed with the other ticket... it is still open, see #3126

File: views/default/page/components/message.php (hunk 4 above)
Comment (jdalsem) [reply views/default/page/components/message.php]: will do

Comment (mrclay) [reply views/default/page/components/message.php]: I looked and found other icon_name $vars keys.

File: views/default/core/js/ui.js (hunk 3 above)
Comment (mrclay) [reply views/default/core/js/ui.js]: Based on this PR I've started to rewrite messages to be more like client-side, where delay is configurable.

File: views/default/page/components/message.php (hunk 4 above)
Comment (jdalsem) [reply views/default/page/components/message.php]: hmmm icon means in a lot of cases something else than just the icon_name... think i will leave it as is for now (icon = html, icon_name is used in elgg_view_icon)

File: views/default/core/js/ui.js (hunk 3 above)
Comment (jdalsem) [reply views/default/core/js/ui.js]: going to merge this... your PR can follow

**Pull request merged**